packages/react/src/List/ListItem.tsx

```diff
@@ -0,0 +1,40 @@
+import React, {type Ref, useContext} from 'react'
+import clsx from 'clsx'
+import {CheckIcon, DashIcon} from '@primer/octicons-react'
+import type {BaseProps} from '../component-helpers'
+import {ListContext} from './UnorderedList'
+
+import styles from './List.module.css'
+
+export type ListItemProps = BaseProps<HTMLElement> & {
+  /**
+   * React.ReactNode and React.ReactNode[] are valid children.
+   */
+  children?: React.ReactNode | React.ReactNode[]
+  /**
+   * The ref object to be attached to the list item.
+   */
+  ref?: Ref<HTMLLIElement>
```

Comment:
BaseProps includes `ref`, you just need to provide this HTML type def as a generic value.

Reply:
In next push